src/components/swap/SwapLineItem.tsx

```diff
@@ -0,0 +1,250 @@
+          if (isPreview) return <Loading />
+          return <>{formatNumber({ input: trade.totalGasUseEstimateUSD, type: NumberType.FiatGasPrice })}</>
+        },
+      }
+    case SwapLineItemTypes.PRICE_IMPACT:
+      // Hides price impact row if the current trade is UniswapX or if we're previewing a trade when the last trade was UniswapX
+      if (isUniswapX || (wasUniswapX && isPreview)) return
+      return {
+        Label: () => <Trans>Price impact</Trans>,
+        TooltipBody: () => <Trans>The impact your trade has on the market price of this pool.</Trans>,
+        Value: () => (isPreview ? <Loading /> : <ColoredPercentRow percent={trade.priceImpact} />),
+      }
+    case SwapLineItemTypes.MAXIMUM_INPUT:
+      if (trade.tradeType === TradeType.EXACT_INPUT) return
+      return {
+        Label: () => <Trans>Maximum input</Trans>,
+        TooltipBody: () => (
+          <Trans>
+            The maximum amount you are guaranteed to spend. If the price slips any further, your transaction will
+            revert.
+          </Trans>
+        ),
+        Value: () => <CurrencyAmountRow amount={trade.maximumAmountIn(allowedSlippage)} />,
+        loaderWidth: 70,
+      }
+    case SwapLineItemTypes.MINIMUM_OUTPUT:
+      if (trade.tradeType === TradeType.EXACT_OUTPUT) return
+      return {
+        Label: () => <Trans>Minimum output</Trans>,
+        TooltipBody: () => (
+          <Trans>
+            The minimum amount you are guaranteed to receive. If the price slips any further, your transaction will
+            revert.
+          </Trans>
+        ),
+        Value: () => <CurrencyAmountRow amount={trade.minimumAmountOut(allowedSlippage)} />,
+        loaderWidth: 70,
+      }
+    case SwapLineItemTypes.EXPECTED_OUTPUT:
+      return {
+        Label: () => <Trans>Expected output</Trans>,
+        TooltipBody: () => (
+          <Trans>
+            The amount you expect to receive at the current market price. You may receive less or more if the market
+            price changes while your transaction is pending.
+          </Trans>
+        ),
+        Value: () => <CurrencyAmountRow amount={trade.postTaxOutputAmount} />,
+        loaderWidth: 65,
+      }
+    case SwapLineItemTypes.ROUTING_INFO:
+      if (isPreview) return { Label: () => <Trans>Order routing</Trans>, Value: () => <Loading /> }
+      return {
+        Label: () => <Trans>Order routing</Trans>,
+        TooltipBody: () => {
+          if (isUniswapX) return <UniswapXDescription />
+          return <SwapRoute data-testid="swap-route-info" trade={trade} syncing={syncing} />
+        },
+        tooltipSize: isUniswapX ? TooltipSize.Small : TooltipSize.Large,
+        Value: () => <RouterLabel trade={trade} />,
+      }
+    case SwapLineItemTypes.INPUT_TOKEN_FEE_ON_TRANSFER:
+    case SwapLineItemTypes.OUTPUT_TOKEN_FEE_ON_TRANSFER:
+      return getFOTLineItem(props)
+  }
+}
+
+function getFOTLineItem(props: SwapLineItemProps): LineItemData | undefined {
```

Comment:
i'd destructure `props` into ` { type, trade }` here so you don't have to do `props.` three times lolz